annubes/task.py

```diff
@@ -142,20 +138,20 @@ def generate_trials(
             `generate_trials()` method's call ("ntrials", "random_state"), and the generated data ("modality_seq",
             "time", "phases", "inputs", "outputs").
         """
-        self._ntrials = ntrials
-
         # Set random state
         if random_seed is None:
             rng = np.random.default_rng(random_seed)
             random_seed = rng.integers(2**32)
         self._rng = np.random.default_rng(random_seed)
         self._random_seed = random_seed
 
+        self._ntrials = self._rng.integers(min(ntrials), max(ntrials)) if isinstance(ntrials, tuple) else ntrials
+
```

Comment:
Good point, this can be clarified! Tuples are conventionally used for intervals, rather than lists. We should restrict the most possible the choices of users in such cases. I took care of such checks in PR #64. There, `ntrials` can only be an integer or a tuple of 2 integers.